# configuration for youbot base joint interface
youbot_base:
  position_frame: odom
  base_frame: base_odom_link
  velocity_topic: cmd_vel
  
  publish_base_state: false #whether the state of the base is to be published on /joint_states as if it was a joint
  publish_rate: 200 # [Hz]
  
  base_controller_ns: base_controller # defines the topic in which position_commands are expected
  base_name: youbot_base #name of the base, joint values are published as three joint values base_name/x base_name/y base_name/theta, following moveit conventions
  
  pid_pos:
    p: 3
    i: 0
    d: 0
  pid_theta:
    p: 6
    i: 0
    d: 0
  
  use_velocity_limits: true  
  unfeasible_rounding_threshold: 
    linear: 0.006 # equal or above gets rounded up to the lowest feasible velocity, everything lower becomes zero
    angular: 0.03
  velocity_limits:
    upper:
      linear:
        min: -0.8
        max: 0.8
      angular:
        min: -2.86
        max: 2.86
    unfeasible:
      linear:
        neg: -0.01
        pos: 0.01
      angular:
        neg: -0.067
        pos: 0.067
      
    
  use_space_limits: false #limit the space in which the youbot may move
  space_limits:
    x:
      min: -5
      max: 5
    y:
      min: -5
      max: 5